### PR TITLE
fix(position): scale receivers proportionally instead of zeroing on insolvent close

### PR DESCRIPTION
**Pull request by joshualyguessennd**
Insolvent closes (full-size liquidation / ADL that can't fully pay) used to credit everything to the pool and zero all receivers , fixed by scaling each receiver's share by the recovered ratio instead of zeroing fees

### CODE REVIEW

**Review by derekawender — APPROVED** [review]

File: contracts/position/DecreasePositionCollateralUtils.sol [hunk 1]
```diff
@@ -817,4 +821,64 @@ library DecreasePositionCollateralUtils {
...
+        if (fees.feeAmountForPool > 0) {
+            MarketUtils.applyDeltaToPoolAmount(
+                params.contracts.dataStore,
+                params.contracts.eventEmitter,
+                params.market,
+                collateralToken,
+                fees.feeAmountForPool.toInt256()
+            );
+        }
+        _distributeTransactionShares(params, fees, collateralToken);
```
Comment (derekawender) [inline contracts/position/DecreasePositionCollateralUtils.sol:876]: There is no check here for if fees are negative or greater than zero. Please ensure that these edge cases do not cause erroneous accounting @joshualyguessennd. Maybe add some tests covering these scenarios

**Review by derekawender — APPROVED** [review]
